docs/desired-state-feedback-specification.md

```diff
@@ -0,0 +1,136 @@
+### Desired State Feedback Representation
+The cloud backend (or OTA update system) needs to keep track of the update progress on both individual level as well as on campaign (or fleet) level. To achieve this purpose, the Update Manager introduces the Desire State Feedback as a common, structured and straightforward way for notifying the backend about the progress and status of the update.
+
+The feedback is comprised by overall information for the current status of the ongoing update process, information for the status for each of the identified components and optional unstructured message. This message is meant to support projects with the need to transfer custom data such as a human readable progress information or internally used status codes.
+Optionally, the feedback can be defined at baseline level by specifying additional information for the baseline the feedback is associated with. If not present, the feedback is considered to be related at domain level.
+
+### Desired State Feedback Data Model
+
+The following table describes all supported properties and sections of the Desired State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| status | string | Status of the Desired State Feedback. The supported statuses are listed below |
+| baseline | string | Title of the baseline, the Desired State Feedback is associated with. Optional |
+| message | string | Info message for the Desired State Feedback |
+| actions | JSON array | Set of Desired State Feedback actions |
+| **Action properties** | | |
+| component | JSON object | The component, which is linked with the action |
+| status | string | Status the of action. The supported statuses are listed below |
+| progress | int | Progress of the action in percentage |
+| message | string | Info message for the action |
+| **Config properties** | | |
+| key | string | Key of the configuration property |
+| value | string | Value of the configuration property |
+| **Component properties** | | |
+| id | string | Identifier of the component |
+| version | string | Version of the component |
+| config | JSON object | Set of component-specific configuration properties as key/value pairs |
+
+### Supported Desired State Feedback statuses
+
+The list of the supported feedback statuses :
+
+| Status | Description |
+| - | - |
+| IDENTIFYING | Denotes that action identification process has started |
+| IDENTIFIED | Denotes that required actions are identified |
+| IDENTIFICATION_FAILED | Denotes some or all of the required actions failed to be indentified |
+| RUNNING | Denotes that identified actions are currently executing |
+| DOWNLOADED_COMPLETED | Denotes that identified actions are downloaded successfully |
+| DOWNLOADED_FAILED | Denotes that not all of the identified actions are downloaded successfully |

```

Comment:
To be consistent with `IDENTIFICATION_FAILED`, I would name this status `DOWNLOAD_FAILED` (and `DOWNLOAD_COMPLETED`).

Reply:
I fully understand your concerns but we do not want to change these constants now.
These constants come from a not-yet-publicly-released Eclipse BfB message specification, that are already implemented in a private backend (Pantaris).
When we have the BfB definitions as open source, we will adapt the implementation inside the Kanto Update Manager respectively. (Perhaps then some of the docs here will become irrelevant/duplicate).